Modules/Scripted/DICOMLib/DICOMProcesses.py

```diff
@@ -392,118 +412,135 @@ def processStdoutLine(self, line):
-                    effectiveServerUrl = "https://demo.kheops.online/api"
-                    from requests.auth import HTTPBasicAuth
-                    from dicomweb_client.session_utils import create_session_from_auth
-                    auth = HTTPBasicAuth('token', token)
-                    session = create_session_from_auth(auth)
-
-            client = DICOMwebClient(url=effectiveServerUrl, session=session, headers=headers)
-
-            # Turn off detailed logging, because it would slow down the file transfer
-            clientLogger = logging.getLogger('dicomweb_client')
-            originalClientLogLevel = clientLogger.level
-            clientLogger.setLevel(logging.WARNING)
-
-            try:
-                for file in self.files:
-                    if not self.progressCallback(f"Sending {file} to {self.destinationUrl.toString()} using {self.protocol}"):
-                        raise UserWarning("Sending was cancelled, upload is incomplete.")
-                    import pydicom
-                    dataset = pydicom.dcmread(file)
-                    client.store_instances(datasets=[dataset])
-            finally:
-                clientLogger.setLevel(originalClientLogLevel)
+    def send(self) -> None:
+        """
+        Sends DICOM files to a remote server. Called on instance initialization.
+        """
+        self.progressCallback(
+            f"Starting send to {self.destinationUrl.toString()} using {self.protocol} protocol"
+        )
 
+        if self.protocol.lower() == "dicomweb":
+            self._sendFilesWithDICOMWeb()
         else:
-            # DIMSE (traditional DICOM networking)
+            self._sendFilesWithDIMSE()
+
+    def _sendFilesWithDIMSE(self) -> None:
+        """
+        Initialize for DIMSE and send files to the remote server.
+
+        :raises UserWarning: if a transfer is cancelled.
+        """
+        # DIMSE (traditional DICOM networking)
+        for file in self.files:
+            self._sendOneFileWithDIMSE(file)
+            if not self.progressCallback(
+                f"Sent {file} to {self.destinationUrl.host()}:{self.destinationUrl.port()}"
+            ):
+                raise UserWarning("Sending was cancelled, upload is incomplete.")
+
+    def _sendFilesWithDICOMWeb(self) -> None:
+        """
+        Initialize for DICOMweb and send files to the remote server.
+
+        :raises ModuleNotFoundError: if `dicomweb_client<0.51` and DICOMweb STOW-RS protocol is requested.
+        :raises UserWarning: if a transfer is cancelled.
+        """
+        # Ensure that correct version of dicomweb-client Python package is installed
+        # pythonweb-client 0.50 was broken (https://github.com/MGHComputationalPathology/dicomweb-client/issues/41)
+        minimumDicomwebClientVersion = "0.51"
+        slicer.util.validate_module_version(
+            "dicomweb_client", minimumDicomwebClientVersion
+        )
```

Comment:
```suggestion
```

The original logic introduced in https://github.com/Slicer/Slicer/commit/b89b3c8a4b2a209ba3654377ee3050da62a14877 and updated in https://github.com/Slicer/Slicer/commit/c25b4e6212849b14f8d2c69324b4ee88b6abd558.

It was intended to check if the version of `dicomweb_client` provided  by Slicer needed to be updated.

Since then, the version has been updated[^1] and this logic can simply be removed.

[^1]: https://github.com/Slicer/Slicer/commits/main/SuperBuild/External_python-dicom-requirements.cmake

Reply:
Thanks for the background. I agree that this check should be unnecessary in the current Slicer build. My goal in this PR has been to introduce authentication paths for DICOMweb integration and otherwise improve code structure with minimal possible changes to functionality/interface. But, if it is more convenient I will remove the unneeded check here.

Reply:
The `dicomweb_client` version check is now removed from `DICOMSender` in this PR.